### PR TITLE
Refresh/Purge API

### PR DESCRIPTION
**Pull request by phg49389**
I've been writing this on-and-off for the past month or so, when all my other projects were waiting on something. This should resolve Issue #2 if we implement it.

There's also some misc Varnish changes that have been developed alongside this branch, but in Puppet. They need to be applied before this functionality will work.

Things still to do: 

- Provide a link from the index page to the R/P API index page. For testing, I was just going to /rpapi directly.
- If we enable Smart Bans in Varnish, then there's a TODO comment in controllers/rpapi.py to support Smart Bans from this app too

### CODE REVIEW

**Comment by phg49389** [comment]
Update: the Varnish-side changes have been implemented, so this will be ready to use as soon as it gets merged in.

**Comment by phg49389** [comment]
Thumbor has been moved into its own module, next to RPAPI. Navbar links to Thumbor and RPAPI have been added. Now there's only one thing left to do:

- If we enable Smart Bans in Varnish, then there's a TODO comment in controllers/rpapi.py to support Smart Bans from this app too

I think this is ready for a code review.

File: app/__init__.py [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
-from bu_cascade.cascade_connector import Cascade
+# from bu_cascade.cascade_connector import Cascade
 
 app = Flask(__name__)
 app.config.from_object('config')
 app.url_map.strict_slashes = False
 
-cascade_connector = Cascade(app.config['SOAP_URL'], app.config['CASCADE_LOGIN'], app.config['SITE_ID'], app.config['STAGING_DESTINATION_ID'])
+# cascade_connector = Cascade(app.config['SOAP_URL'], app.config['CASCADE_LOGIN'],
+#                             app.config['SITE_ID'], app.config['STAGING_DESTINATION_ID'])
```
Comment (c-schwarze) [inline app/__init__.py]: Do we not need this anymore? If we don't, we should delete it

**Review by c-schwarze — COMMENTED** [review]
As stated in the github issue, we might be moving away from using varnish to cache images and things. If that is the case, I think what you are adding here will be removed. I'm not sure that we should add this in. We can chat more about this, if you would like

File: app/__init__.py (hunk 1 above)
Comment (phg49389) [reply app/__init__.py]: I think this was initially imported so we could use Cascade groups to determine who should be allowed to use this app, but then we migrated over to IAM groups via LDAP. I wasn't able to find any usages of `cascade_connector` in Cache Clear, so I just commented it out.

I agree though: if it's not used, it should be removed. I'll go do that.

**Comment by phg49389** [comment]
I understand that we're not caching images in Varnish right now, but we're still caching static files like CSS, JS, and some font files, as well as certain routes in business-signage.bethel.edu and hall-of-fame.bethel.edu. Additionally, it's not unreasonable to anticipate that images could be cached once again. If I recall correctly, the reason we stopped caching images was due to triple-caching issues between Varnish and Thumbor. Thus, we only need to prevent cdn.bethel.edu/resize routes from being cached and the rest of images would be eligible for caching once more.

Setting aside the potential utility of this PR, this would complete the second half of the Refresh/Purge API that has been in Varnish for almost 11 months now. This PR has been planned for a long time, and I want to see it through even if it's not used right away. It is logical for me to write it since I authored the other half, and this prevents an awkward situation where someone else has to read through the VCL source code to understand how it works and read through issue 2 to see what I had in mind. A sort of "picking up where someone else left off" kind of situation, if you will.

**Comment by c-schwarze** [comment]
👍 Alright, sounds good.

**Review by c-schwarze — APPROVED** [review]

**Pull request merged**